### PR TITLE
Add moist-baroclinic-wave benchmark case to GCM experiments

### PR DESCRIPTION
**Pull request by bischtob**
# Description

This adds the Moist Baroclinic Wave DCMIP 2016 benchmark to the GCM experiments.

co-authored by: @LenkaNovak & @bischtob 

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://CliMA.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: experiments/AtmosGCM/moist-baroclinic-wave.jl [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+    # Set up the atmosphere model
+    exp_name = "MoistBaroclinicWave"
+
+    model = AtmosModel{FT}(
+        AtmosGCMConfigType,
+        param_set;
+        ref_state = ref_state,
+        turbulence = ConstantViscosityWithDivergence(FT(0.0)),
+        moisture = DryModel(),
```
Comment (akshaysridhar) [inline experiments/AtmosGCM/moist-baroclinic-wave.jl:123]: Problem with moisture requires specification of `EquilMoist` (if expecting to use saturation adjustment) and definition of the `state.moisture.\rhoq_tot` variable (line 104).

File: experiments/AtmosGCM/moist-baroclinic-wave.jl [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+    # Set up diagnostics
+    dgn_config = config_diagnostics(FT, driver_config)
+
+    # Set up user-defined callbacks
+    filterorder = 64
+    filter = ExponentialFilter(solver_config.dg.grid, 0, filterorder)
+    cbfilter = GenericCallbacks.EveryXSimulationSteps(1) do
+        @views begin
+          solver_config.Q.data[:, 1, :] .-= solver_config.dg.state_auxiliary.data[:, 8, :]
```
Comment (akshaysridhar) [inline experiments/AtmosGCM/moist-baroclinic-wave.jl:200]: should be able to use vars aware MPIStateArrays here to identify state indices with their `@vars names`

Comment (thomasgibson) [reply experiments/AtmosGCM/moist-baroclinic-wave.jl:200]: This should hopefully get cleaned up once #1200 and #1182 land

File: experiments/AtmosGCM/moist-baroclinic-wave.jl [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    τ_1::FT = 1/T_0 * τ_z_1 + (T_0-T_P)/T_0/T_P * τ_z_2 * τ_z_3 
+    τ_2::FT = 0.5 * (K+2) * (T_E-T_P)/T_E/T_P * τ_z_2 * τ_z_3 
+    τ_int_1::FT = 1/Γ * (τ_z_1-1) + z * (T_0-T_P)/T_0/T_P * τ_z_2 
+    τ_int_2::FT = 0.5 * (K+2) * (T_E-T_P)/T_E/T_P * z * τ_z_3
+
+    # temperature, pressure, specific humidity, density
+    T_v::FT = 1 / (τ_1 - τ_2 * I_T) 
+    p::FT = p_0 * exp(-_grav/_R_d * (τ_int_1 - τ_int_2 * I_T))
+    η::FT = p/p_0
+    #q::FT = q_0 * exp(-(φ/φ_w)^4) * exp(-((η-1)*p_0/p_w)^2)
```
Comment (akshaysridhar) [inline experiments/AtmosGCM/moist-baroclinic-wave.jl:71]: I suppose this is because you want to run the dry case first ?

Comment (bischtob) [reply experiments/AtmosGCM/moist-baroclinic-wave.jl:71]: :)
correct

File: experiments/AtmosGCM/moist-baroclinic-wave.jl [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    state.ρ = ρ
+    state.ρu = ρ * SVector{3, FT}(-sin(λ)*u, cos(λ)*u, 0) 
+    state.ρe = ρ * total_energy(bl.param_set, e_kin, e_pot, T)
+    
+    nothing
+end
+
+function config_moist_baroclinic_wave(FT, poly_order, resolution)
+    # Set up a reference state for linearization of equations
+    temp_profile_ref = DecayingTemperatureProfile{FT}(param_set, FT(275), FT(75), FT(45e3))
```
Comment (charleskawczynski) [inline experiments/AtmosGCM/moist-baroclinic-wave.jl:109]: Is `FT(75)` correct? Seems low

**Comment by thomasgibson** [comment]
Superseded by #1340

**Pull request closed without merging**